magma/primitives/when.py

```diff
@@ -139,6 +139,7 @@ def __init__(self, block: WhenBlock):
         self._set_definition_attr(_ISWHEN_KEY, True)
         self._set_definition_attr("_builder_", self)
         self._is_when_builder_ = True
+        self._set_definition_attr("_is_when_builder_", True)
```

Comment:
let's add a separate free function `is_when_builder` which is just `def is_when_builder(obj) -> bool: return getattr(obj, "_is_when_builder_", False)`

can also additionally abstract `"_is_when_builder_"` key into `_IS_WHEN_BUILDER_KEY` constant at the top.